tests/unit/conftest.py

```diff
@@ -80,3 +83,37 @@ def acc_client():
     acc = create_autospec(AccountClient)  # pylint: disable=mock-no-usage
     acc.config = Config(host="https://accounts.cloud.databricks.com", account_id="123", token="123")
     return acc
+
+
+class MockPathLookup(PathLookup):
+    def __init__(self, cwd='source_code/samples', sys_paths: list[Path] | None = None):
+        super().__init__(Path(__file__).parent / cwd, sys_paths or [])
+
+    def change_directory(self, new_working_directory: Path) -> 'MockPathLookup':
+        return MockPathLookup(new_working_directory, self._sys_paths)
+
+    def resolve(self, path: Path) -> Path | None:
+        candidates = [path]
+        if not path.name.endswith('.txt'):
+            candidates.append(Path(f"{path}.txt"))
+        for candidate in candidates:
+            absolute_path = super().resolve(candidate)
+            if not absolute_path:
+                continue
+            return absolute_path
+        return None
+
+    def __repr__(self):
+        return f"<MockPathLookup {self._cwd}>"
+
+
+@pytest.fixture
+def path_lookup():
+    return MockPathLookup()
+
+
+@pytest.fixture
+def notebook_resolver():
```

Comment:
```suggestion
def mock_notebook_resolver():
```

Reply:
let's stick to Serge's naming convention, the fact that it's a mock is a hidden detail and should remain so

Reply:
The convention is to prefix `mock_` right, e.g. `mock_installation`

Reply:
done. poor convention imho